app/models/product.rb

```diff
@@ -498,6 +498,7 @@ def set_initial_values
   # Try to find the best name for the new products
   def choose_default_name
     return if name.present?
+    return if name.blank?
```

Comment:
```
return if name.present?
return if name.blank?
```
is equivalent to have deleted replaced the whole method by a simple `return`, this would break behavior in all the other places this method is called.

Reply:
@Aquaj 
Here Previously I used self.class == Product  This condition return  true in our case only, when we create the Product, Can I used this?
So our new condition will be like that

return if name.present?
return if name.blank? && self.class == Product

Its okay